File: algorithms/algorithms-cplusplus/leetcode/Question_0771_Jewels_and_Stones.cc
#include <string>
#include <vector>

using namespace std;

class Solution {
 public:
  int numJewelsInStones(string jewels, string stones) {
    vector<bool> isJewels(256, false);
    for (char c: jewels) {
      isJewels[c] = true;
    }
    int res = 0;
    for (char c : stones) {
      if (isJewels[c]) {
        res++;
      }
    }
    return res;
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0771_Jewels_and_Stones.java
package com.lanceolata.coding.algorithms.java.leetcode;

import java.util.Arrays;

public class Question_0771_Jewels_and_Stones {
    public int numJewelsInStones(String jewels, String stones) {
        boolean[] isJewels = new boolean[256];
        Arrays.fill(isJewels, false);
        for (char c: jewels.toCharArray()) {
            isJewels[c] = true;
        }
        int res = 0;
        for (char c : stones.toCharArray()) {
            if (isJewels[c]) {
                res++;
            }
        }
        return res;
    }
}
